src/cli.ts

```diff
@@ -74,11 +82,40 @@ export interface CliOptions {
   orderBy: 'depender' | 'dependee';
   findSolutions: boolean;
   install: boolean;
+  config: string;
 }
 
 if (options.help) {
   process.exit(-2);
 }
 
-const packageManager = getPackageManager(options.yarn, options.npm);
-checkPeerDependencies(packageManager, options);
+async function isOnIgnoredBranch(ignoredBranchPrefixes: string[]) {
+  const currentBranch = await getCurrentBranch();
+  return ignoredBranchPrefixes.some(prefix => currentBranch.startsWith(prefix));
+}
+
+async function isFeatureFlagOff(launchdarklyClientId: string) {
+  const isPeerDepCheckOn = await getPeerDepCheckFF(launchdarklyClientId);
+  return !isPeerDepCheckOn;
+}
+
+async function main() {
+  const { ignoredBranchPrefixes, launchdarklyClientId } = getConfig(options.config);
+
+  if (ignoredBranchPrefixes && await isOnIgnoredBranch(ignoredBranchPrefixes)) {
```

Comment:
This is mainly for skipping the check in PI builds on `atlaskit-branch-deploy` branches. In this way, it's guaranteed that product integrator won't get affected and we also don't need to install extra packages to satisfy the check when dogfooing product fabric on master.

Reply:
This should be done outside of this package. This package responsibility is to check for peer dependencies. whether it runs the check or not should be determined by the caller.

Reply:
hmm but the caller is JFE and CFE. I guess I'll need to create another package `check-peer-dependencies-if-needed` that contains the feature flag or branch check, which then depends on `check-peer-dependencies`.

could this get overcomplicated? If we wanted to fix issues or add features, we'd publish both `check-peer-dependencies` and `check-peer-dependencies-if-needed` for products to use.